core/main.go

```diff
@@ -210,13 +210,13 @@ func (a *AutoSpotting) convertRawEventToCloudwatchEvent(event *json.RawMessage)
 
 // parse instance events and execute the relative methods
 func (a *AutoSpotting) processEventInstance(eventType string, region string, instanceID *string, instanceState *string) error {
-	if eventType == "IS" {
+	if eventType == InstanceStateChangeNotificationCode {
 		// If event is Instance state change
 		if len(a.config.sqsReceiptHandle) != 0 {
 			logger.SetPrefix(fmt.Sprintf("SQ:%s ", *instanceID))
 		}
 		a.handleNewInstanceLaunch(region, *instanceID, *instanceState)
-	} else if eventType == "II" || eventType == "IR" {
+	} else if eventType == SpotInstanceInterruptionWarningCode || InstanceRebalanceRecommendationCode == "IR" {
```

Comment:
I haven't tested this but I think this condition might be incorrect

Reply:
sorry, was a typo ...i fix it

Reply:
No problem, this is why we have code reviews 😁